azurelinuxagent/ga/monitor.py

```diff
@@ -36,58 +36,148 @@
 from azurelinuxagent.ga.periodic_operation import PeriodicOperation
 
 
-def generate_extension_metrics_telemetry_dictionary(schema_version=1.0,
-                                                    performance_metrics=None):
-    if schema_version == 1.0:
-        telemetry_dict = {"SchemaVersion": 1.0}
-        if performance_metrics:
-            telemetry_dict["PerfMetrics"] = performance_metrics
-        return telemetry_dict
-    else:
-        return None
-
-
 def get_monitor_handler():
     return MonitorHandler()
 
 
+class PollResourceUsageOperation(PeriodicOperation):
```

Comment:
moved the periodic tasks into their own classes, which can be used/tested independently of the monitor thread